SoftLayer/CLI/formatting.py

```diff
@@ -440,3 +444,29 @@ def _format_list_objects(result):
         table.add_row(values)
 
     return table
+
+
+def csv_output_format(data, delimiter=','):
+    """Formating a table to csv format and show it."""
+    data = clean_null_table_rows(data)
+    with tempfile.TemporaryDirectory() as temp_file:
+        f_name = os.path.join(temp_file, 'temp_csv_file')
+        with open(f_name, 'w', encoding='UTF8') as file:
+            writer = csv.writer(file, delimiter=delimiter)
```

Comment:
I think you can just pass `sys.stdout` as the filename to print the result to the console. Don't create a file just to read from it again.
```suggestion
            writer = csv.writer(sys.stdout, delimiter=delimiter)
```

Reply:
Used sys.stdout for print csv datas
Added some unit test
Supported nested tables to csv outputformat